app/controllers/OrderController.java

```diff
@@ -1,31 +1,35 @@
 package controllers;
 
-import io.sphere.client.SphereClientException;
-import io.sphere.client.exceptions.SphereException;
-import io.sphere.client.shop.model.Cart;
-import io.sphere.client.shop.model.Product;
-import io.sphere.client.shop.model.Variant;
+import io.sphere.sdk.carts.Cart;
+import io.sphere.sdk.products.ProductVariant;
 import models.OrderPageData;
 import play.Configuration;
 import play.Logger;
 import play.mvc.Result;
-import sphere.Sphere;
+import services.CartService;
 import views.html.order;
 import views.html.success;
 
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
 public class OrderController extends BaseController {
 
-    public OrderController(final Sphere sphere, final Configuration configuration, final Product product) {
-        super(sphere, configuration, product);
+    private final CartService cartService;
+
+    public OrderController(final Configuration configuration, final CartService cartService) {
+        super(configuration);
+        this.cartService = requireNonNull(cartService, "'cartService' must not be null");
     }
 
     public Result show() {
-        final Cart cart = sphere().currentCart().fetch();
-        if (cart.getLineItems().size() > 0) {
-            final int selectedFrequency = frequency(cart.getId());
+        final Cart currentCart = cartService.getOrCreateCart(session());
+        if (currentCart.getLineItems().size() > 0) {
```

Comment:
`!currentCart.getLineItems().isEmpty()` is better for list implementations that are huge and do not cache the size. Also I think it is better readable.